src/test/java/com/devonfw/sample/archunit/ComponentRuleTest.java

```diff
@@ -0,0 +1,263 @@
+                    boolean isAllowedDependency = isComponentBatchLayerCompliant(sourceClass, targetClass);
+
+                    // WARNING: Dependency of a components batch layer towards another components
+                    // layer other than logic wont be registered as a violation.
+                    // (Other rules cover these violations.)
+                    if (targetPkg.isLayerBatch() && !isAllowedDependency) {
+                        String message = String.format(
+                                "'%s.%s' is dependend on '%s.%s'. Violated in: (%s). Dependency towards (%s)",
+                                sourceClassComponent, sourceClassLayer, targetClassComponent, targetClassLayer,
+                                sourceClass.getDescription(), targetClass.getDescription());
+                        events.add(new SimpleConditionEvent(sourceClass, true, message));
+                    }
+                }
+            }
+        }
+    };
+
+    /**
+     * verifying that the batch layer of a component may not depend on the
+     * logic layer of another component.
+     */
+    @ArchTest
+    public static final ArchRule C7_no_dependencies_from_a_components_batch_layer_to_anothers_component_logic_layer = noClasses()
+            .should(haveComponentDataaccessLayerDependingOnDiffComponentsDataaccessLayer)
+            .as("Code from batch layer of a component shall not depend on logic layer of a different component.")
+            .allowEmptyShould(true);
+
+    /**
+     * Dependency of a components batch layer towards the same components logic or
+     * common layer is allowed.
+     */
+    public static boolean isComponentBatchLayerCompliant(JavaClass sourceClass, JavaClass targetClass) {
+        boolean isAllowed = false;
+        PackageStructure sourcePkg = PackageStructure.of(sourceClass.getPackageName());
+        PackageStructure targetPkg = PackageStructure.of(targetClass.getPackageName());
+        // Components batch layer can depend on their own lower layers: logic and
+        // common.
+        if (sourcePkg.hasSameComponent(targetPkg)
+                && (targetPkg.isLayerLogic() || targetPkg.isLayerCommon())) {
+            isAllowed = true;
+        }
+        // Components may always depend on the default business component.
+        if (targetPkg.getComponent().equals("general")
+                && (targetPkg.isLayerLogic() || targetPkg.isLayerCommon())) {
+            isAllowed = true;
+        }
+        return isAllowed;
+    }
+
+    /**
+     * Dependency of a components logic layer towards the same components dataaccess
+     * or common layer is allowed.
+     */
+    public static boolean isComponentLogicLayerCompliant(JavaClass sourceClass, JavaClass targetClass) {
+        boolean isAllowed = false;
+        PackageStructure sourcePkg = PackageStructure.of(sourceClass.getPackageName());
+        PackageStructure targetPkg = PackageStructure.of(targetClass.getPackageName());
```

Comment:
Same here:
```
private static boolean isComponentLogicLayerCompliant(PackageStructure sourcePkg, PackageStructure targetPkg) {

Reply:
Changed method to `private`.
Readjusted the method signature to use `PackageStructure` types.